AIEnrichmentPipeline/infra/videoindexer/scripts/videoindexer.ps1

```diff
@@ -0,0 +1,92 @@
+param([String]$type)
+
+if ($ENV:debug_log) {
+    Start-Transcript -Path "./videoindexer.$type.log"
+}
+
+# Terraform provider sends in current state
+# as a json object to stdin
+$stdin = $input
+$location = $ENV:LOCATION
+$apiKey = $ENV:API_KEY
+$createJSON = $ENV:CREATE_JSON
+
+$VIBaseUrl = "https://api.videoindexer.ai"
+$headers = @{
+    "Ocp-Apim-Subscription-Key" = "$apiKey"
+}
+$userTokenEndpoint = "$VIBaseUrl/Auth/trial/Users/me/AccessToken?allowEdit=true"
+$accountTokenEndpoint = "$VIBaseUrl/Auth/$location/Accounts?generateAccessTokens=false&allowEdit=false"
```

Comment:
[Optional] Making sure you are aware it's a read only access token (allowEdit = false)
Also there is a newer API https://api-portal.videoindexer.ai/api-details#api=Operations&operation=Get-Account-Access-Token-With-Permission which does the same thing and have fine grained roles:

Reader (allows reading content) / Contributor (allows modifying content) / MyAccessAdministrator (allows managing the acting user's permission) / Owner (allows managing permissions and modifying content)